### ISSUE TITLE
Annotations Not Loading v2.0

### ISSUE THREAD

**Issue by travisgalloway** [openannotation/annotator#579]
I'm implementing Annotator with epub.js. I'm able to create new annotations in the browser and write them to storage. They appear highlighted as normal, but fail to reload when I navigate away from then back to the page. The annotation.load() returns OPTIONS 200 and GET 302 with the correct annotations in JSON, but nothing is rendered on the page. 

Here's the Annotator calls compiled from Coffeescript

``` javascript
    setAnnotation = function() {
      return {
        beforeAnnotationCreated: function(annotation) {
          annotation.book_id = bookID;
          annotation.base_uri = baseURI;
        },
        annotationCreated: function(annotation) {
          return console.log(annotation);
        }
      };
    };
    app = new annotator.App;
    app.include(annotator.ui.main);
    app.include(annotator.storage.http, {
      prefix: 'http://localhost:3000/api',
      headers: localHeaders
    });
    app.include(setAnnotation);
    return app.start().then(function() {
      app.annotations.load({
        book_id: bookID,
        base_uri: baseURI
      });
    });
```

Here's the JSON response

``` json
[
    {
        "id": 2,
        "user_id": 1,
        "book_id": 1,
        "base_uri": "http://localhost:3000/books/42c4d399-f989-49f1-bcb2-fff825a3dace/extract/OPS/chapter_002.xhtml",
        "quote": "Now having a night, a day, and still another night following before me in New Bedford, ere I could embark for my destined port, it became a matter of concernment where I was to eat and sleep meanwhile. It was a very dubious-looking, nay, a very dark and dismal night, bitingly cold and cheerless. I knew no one in the place. With anxious grapnels I had sounded my pocket, and only brought up a few pieces of silver,—So, wherever you go, Ishmael, said I to myself, as I stood in the middle of a dreary street shouldering my bag, and comparing the gloom towards the north with the darkness towards the south—wherever in your wisdom you may conclude to lodge for the night, my dear Ishmael, be sure to inquire the price, and don’t be too particular.",
        "ranges": [
            {
                "start": "/section[1]/p[3]/span[1]",
                "startOffset": 0,
                "end": "/section[1]/p[4]/span[1]",
                "endOffset": 0
            }
        ],
        "text": "Chapter 2 - 2",
        "created_at": "2015-11-23T19:52:36.011Z",
        "updated_at": "2015-11-23T19:52:36.011Z"
    }
]
```

Any thoughts?

**Comment by TomThak** [comment]
I have quite similiar issue. I use store plug in to prepare data. 
I make annotation, it calls by ajax my phtml script on the server after pressing save.
It does the POST on firebug console, I can see in POST card the JSON data, but it returns JS error with:
TypeError: data is null
}else if(instance){return options&&instance[options].apply(instance,args)}else{i...

but in POST there are data... as JSON, so I dont know what data it expects, I copied the example from the documentation.

However it passes the data to my script so I can save them for example into file.
When I reload the page it performs search by "loadFromSearch".
I can see in respone of GET ajax request on firebug console, that there is JSON respone
for example:

{"ranges":[{"start":"\/p[3]\/b[1]","startOffset":59,"end":"\/p[3]","endOffset":262}],"quote":"doby","text"
:"dfsdg","uri":"uriofthisdocument"}

but it doesn't load / highlight it into the page.

What was also strange to me - if I try to save it, it POST the data in JSON, but there is not "ID" of the annotation included in the data.

I'm having this format in POST: 

{"ranges":[{"start":"/p[4]","startOffset":0,"end":"/p[4]","endOffset":6}],"quote":"Pravo","text":"adfgfadg"
,"uri":"uriofthisdocument"}

Just ranges, quote, text and URI, but no ID. But in the documentation there is at least ID to be able to recognize and update / delete the annotation.

Any ideas?

**Comment by TomThak** [comment]
I found, where is the mistake - I tried to track how it works on annotatorjs.org site, and they use "rows" like a container of json element. If I tried my simple demo and I added rows into my simple example,
it starts working and starts highlighting and displaying. 

They sould add it into the documentation, that the input JSON format of the data (create POST) is not the same like expected JSON data for loadFromSearch function. It expects to have rows there.

Here is an example of the structure, which works for me:

```
{
  "rows":[
    {
      "ranges":[
        {"start":"/p[6]","startOffset":368,"end":"/p[6]","endOffset":708}
      ],
      "quote": "my quote",
      "text":"mytext",
      "uri":"myuri",
      "id":"mygeneratedid"
    }  
  ],
  "total": mycountoftherows(integer)
}
```

When it is not "packed" in "rows:"[] the LoadFromSearch doesn't load the annotations into the page.
